### PR TITLE
Get `CorrelatedStack` data as `np.ndarray`

### PR DESCRIPTION
**Pull request by rpauszek**
**Why this PR?**
Sometimes you just want the pure data to work on. This adds a convenience method so that users don't have to fiddle around with the internal source data classes.

[docs are here](https://lumicks-pylake.readthedocs.io/en/wit_get_data/tutorial/correlatedstacks.html)

### CODE REVIEW

File: docs/tutorial/correlatedstacks.rst [hunk 1]
```diff
@@ -24,6 +24,14 @@ from Bluelake if available. This functionality can be turned off with the option
 
     stack2 = lk.CorrelatedStack("example2.tiff", align=False)
 
+You can obtain the image stack data as a `numpy` array using the `get_image()` method::
+
+    red_data = stack.get_image(channel="red") # shape = [y_pixels, x_pixels, n_frames]
+    rgb_data = stack.get_image(channel="rgb") # shape = [y_pixels, x_pixels, 3 channels, n_frames]
```
Comment (JoepVanlier) [inline docs/tutorial/correlatedstacks.rst:30]: Is there a reason to deviate from `Scan` here? Multiframe scans are stored as `n_frames`, `pixel`, `pixel`, `colors`.

File: lumicks/pylake/detail/widefield.py [hunk 2]
```diff
@@ -193,6 +193,11 @@ class ImageDescription:
     def __init__(self, tiff_file, align_requested):
         first_page = tiff_file.pages[0]
         self.is_rgb = first_page.tags["SamplesPerPixel"].value == 3
+        self._cmap = (
+            {"red": 0, "green": 1, "blue": 2, "rgb": slice(None), None: slice(None)}
+            if self.is_rgb
+            else {None: slice(None)}
```
Comment (JoepVanlier) [inline lumicks/pylake/detail/widefield.py:199]: Can't this just be an empty dict?

Or even just `None`? (Considering you have the if-statement down there anyway and alignment matrices only exist in multi-color images?).

File: lumicks/pylake/correlated_stack.py [hunk 3]
```diff
@@ -115,6 +115,26 @@ def crop_by_pixels(self, x_min, x_max, y_min, y_max):
...
 
+    def get_image(self, channel=None):
+        """Get image data for the full stack as an `np.ndarray`.
+
+        Parameters
+        ----------
+        channel : {'red', 'green', 'blue', 'rgb', `None`}
+            The color channel of the requested data.
+            For single-color data, this argument is ignored.
+            For RGB data the default value `None` corresponds to 'rgb'.
```
Comment (JoepVanlier) [inline lumicks/pylake/correlated_stack.py:126]: Can't we just make the default `"rgb"` and ignore the channel argument when it's a single color stack?

It doesn't seem like you have error handling for requesting `rgb` when there is no `rgb` anyhow?

Then you also don't need the `None` entry anymore in the `_cmap.`

**Review by JoepVanlier — COMMENTED** [review]
Cool!

It's good to have this back up to parity.

Please also update the deprecation notice. Now it says the following:
`All operations on these objects should be handled through the `CorrelatedStack` public API`
But it would be good to give specific instructions now that this function is available.

File: docs/tutorial/correlatedstacks.rst (hunk 1 above)
Comment (rpauszek) [reply docs/tutorial/correlatedstacks.rst]: the reason is that I meant to make sure it was in line with `Scan` but somehow permuted through every other combination of axes before finally getting it right.....

File: lumicks/pylake/detail/widefield.py (hunk 2 above)
Comment (rpauszek) [reply lumicks/pylake/detail/widefield.py]: agreed. and having the axes in the correct order, the reasons for making a more complicated handling basically went away, so we're back to the original code for this

File: lumicks/pylake/correlated_stack.py [hunk 4]
```diff
@@ -115,6 +115,23 @@ def crop_by_pixels(self, x_min, x_max, y_min, y_max):
...
+            The color channel of the requested data.
+            For single-color data, this argument is ignored.
+        """
+        if self.src._description.is_rgb:
+            channel_indices = {"red": 0, "green": 1, "blue": 2, "rgb": slice(None)}
+            slc = (slice(None), slice(None), channel_indices[channel])
+        else:
+            slc = (slice(None),)
+
+        return np.stack([frame.data[slc] for frame in self], axis=0)
```
Comment (JoepVanlier) [inline lumicks/pylake/correlated_stack.py:133]: Correct me if I'm wrong, but I think you need to squeeze this one to get rid of the extraenous axis (if you want to make it fully consistent with `Scan`)?

I found it when trying to slice the stack. Might also be good to add a test for this:
```
np.testing.assert_array_equal(np.squeeze(stack[0].get_image()), rgb_data)
```

**Review by JoepVanlier — COMMENTED** [review]
Mostly looks good to me. Just one minor thing remaining when it comes to compatibility with the `Scan` API.

**Review by JoepVanlier — APPROVED** [review]
LGTM! 👍

**Pull request merged**